lib/pure/net.nim

```diff
@@ -121,6 +121,7 @@ const
 type
   SocketImpl* = object ## socket type
     fd: SocketHandle
+    closed: bool
```

Comment:
instead of introducing another state, i would rather reuse `fd` by setting it to invalid value.